std/format/internal/floats.d

```diff
@@ -18,8 +18,21 @@ import std.format.spec : FormatSpec;
 
 package(std.format) enum RoundingMode { up, down, toZero, toNearestTiesToEven, toNearestTiesAwayFromZero }
 
-package(std.format) auto printFloat(T, Char)(return char[] buf, T val, FormatSpec!Char f,
-                                             RoundingMode rm = RoundingMode.toNearestTiesToEven)
+// wrapper for unittests
+private auto printFloat(T, Char)(return char[] buf, T val, FormatSpec!Char f,
+                                 RoundingMode rm = RoundingMode.toNearestTiesToEven)
+if (is(T == float) || is(T == double)
+    || (is(T == real) && (T.mant_dig == double.mant_dig || T.mant_dig == 64)))
+{
+    import std.array : appender;
+    auto w = appender!string();
+
+    auto result = printFloat(buf, w, val, f, rm);
+    return result ~ w.data;
+}
+
+package(std.format) auto printFloat(Writer, T, Char)(return char[] buf, auto ref Writer w, T val,
+    FormatSpec!Char f, RoundingMode rm = RoundingMode.toNearestTiesToEven)
```

Comment:
Shouldn't this be `@nogc` then?

Reply:
Not yet. It's only `nan` and `inf`. All other values do still allocate. (The plan is to make 0 @nogc in the next PR (needs a change to `writeAligned`). And then to add three more PRs about `%a`, `%e` and `%f`, which can be done in parallel. When all these PRs are through, it's completely `@nogc`.)

(Aside from this, I thought, that templates should not be marked with `@nogc` because that is inferred automatically?)

Reply:
> Not yet. It's only nan and inf. All other values do still allocate. (The plan is to make 0 @nogc in the next PR (needs a change to writeAligned). And then to add three more PRs about %a, %e and %f, which can be done in parallel. When all these PRs are through, it's completely @nogc.)

This should be stated in the description of this PR and all subsequent PRs need to link to this one so that reviewers can easily grasp what is happening here.

> (Aside from this, I thought, that templates should not be marked with @nogc because that is inferred automatically?)

That is true, however, for templates that of which all instantiations are `@nogc`, the attribute can be slapped there. Of course, if the template calls a user defined function then it cannot be made `@nogc`, but in this case we can slap `@nogc` to unittests.

Reply:
> This should be stated in the description of this PR and all subsequent PRs need to link to this one so that reviewers can easily grasp what is happening here.

Sorry. I didn't know that, when I wrote that PR at friday. At that time I just tried to find a start into making floating points `@nogc`. During the weekend I checked out how to continue and finally ended with the four followup PRs. It didn't occur to me, that I should have added this here. (I'll do so in a moment though.)